### PR TITLE
Check HTTP Response status when trying to download minikube.iso

### PR DESCRIPTION
**Pull request by r2d4**
Check the response so that we don't actually download an error page or
something other than the ISO we're looking for.

This might catch some (maybe not all) of the errors we've seen with #471

### CODE REVIEW

File: pkg/minikube/cluster/cluster.go [hunk 1]
```diff
@@ -308,6 +308,8 @@ func (m *MachineConfig) CacheMinikubeISOFromURL() error {
 	response, err := http.Get(m.MinikubeISO)
 	if err != nil {
 		return err
+	} else if response.StatusCode != http.StatusOK {
```
Comment (jimmidyson) [inline pkg/minikube/cluster/cluster.go]: This whole conditional block could be reformatted to reduce the indentations as something like (also defer closing response body should be done ASAP after checking GET error):

```go
response, err := http.Get(m.MinikubeISO)
if err != nil {
	return err
}

defer response.Body.Close()

if response.StatusCode != http.StatusOK {
	return fmt.Errorf("Received %d response from %s while trying to download minikube.iso", response.StatusCode, m.MinikubeISO)
}

out, err := os.Create(m.GetISOCacheFilepath())
if err != nil {
	return err
}
defer out.Close()
if _, err = io.Copy(out, response.Body); err != nil {
	return err
}

return nil

**Comment by jimmidyson** [comment]
Nit on formatting (not a problem with the PR). This is a good first step, we should do proper checksum validation too, but this is good enough for now.

**Comment by dlorenc** [comment]
For checksum validation we could just use a constant (like we do with the minikube ISO URL) for now, but i agree that can come in another PR.

**Comment by jimmidyson** [comment]
We should have the checksums in the release JSON files now shouldn't we? Or we can use a convention of <filename>.sha256 or whatever to try to see if it exists when download files? Hard-coding is a bit annoying/rubbish. Anyway, conversation for another issue/PR.

**Comment by r2d4** [comment]
Thanks for the comments @jimmidyson - I think its a lot clearer the way you suggested.  I refactored that conditional block.

**Comment by dlorenc** [comment]
LGTM

**Pull request merged**